examples/framework/preferences/minimalEditor/schemas/primary.js

```diff
@@ -0,0 +1,37 @@
+ /*
+Copyright 2015 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+/* global fluid */
+
+(function () {
+    "use strict";
+
+   /**
+     * Primary Schema
+     * This schema defines the preference(s) edited by this preferences editor:
+     * their names, types, default values, etc.
+     */
+    fluid.defaults("minEditor.primarySchema", {
```

Comment:
Yes, this is (somewhat) ok - the relevant discussion on this point about the meaning of names within these schemas is at http://lists.idrc.ocad.ca/pipermail/fluid-work/2013-June/009117.html . Note that the interpretation of names like "minEditor.heatedSeats" is meant to represent what we called a "common grade name" - and if "minEditor.primarySchema" can be a grade name, "minEditor.heatedSeats" can be one too, although we should continue to warn people that they should use something resembling an organisation or community name as the first path component of one of these grade names. In general it's ok for a primary schema grade to advertise the schemas for multiple "common grades". This is not necessarily a best practice, depending on how a community is planning to organise its preferences. If all it thinks it is doing is "writing an editor" then this is ok - but bear in mind that our future plans are for preferences editors to be driven by a UI which can draw upon the primary and aux schemas of multiple communities to aggregate together a preferences editor. In such a world, having multiple bits of schema fused into a single grade will be a bit of an annoyance, but not a fatal one.